docs/DeveloperGuide.md

```diff
@@ -1,38 +1,308 @@
+| v1.0    | user                  | be able to delete modules that I am no longer interested in from my list | ignore them in my planning                                                |
+| v1.0    | user                  | edit my previous modules that I have added                               | make changes to my plans whenever needed                                  |
+| v1.0    | user                  | be able to view my past modules that I have saved                        | refer back to my previous lists                                           |
+| v2.0    | user                  | see the list of valid commands                                           | review the valid commands                                                 |
+| v2.0    | potential SEP student | be able to view the full list of modules available                       | consider all my available options                                         |
+| v2.0    | potential SEP student | be able to view the full list of partner universities available          | consider all my available options                                         |
+| v2.0    | user                  | be able to view all module mappings for an NUS module code               | consider modules to map                                                   |
+| v2.0    | user                  | be able to view all module mappings offered by a partner university      | consider modules to map                                                   |
+| v2.0    | user                  | see my most 5 recent deleted modules                                     | check my history                                                          |
+| v2.0    | user                  | be able to favourite my lists                                            | keep that of my priorities                                                |
+| v2.0    | user                  | various search functions for my lists                                    | search for information faster                                             |
+| v2.0    | user                  | be able to add class timings and create a timetable for SEP              | keep track of my timetable                                                |
+| v2.1    | new user              | use a program that does not fail unexpectedly                            | do not have to deal with handling crashes                                 |
+
 
-## Non-Functional Requirements
+## 3. Non-Functional Requirements
 
-{Give non-functional requirements}
+1. Java 11 or above installed
+2. Program built to support only single user
+3. No internet connection necessary
 
-## Glossary
+## 4. Glossary
 
-* *glossary item* - Definition
+- HU: home university
+- PU: partner university
+- u/ - university
+- m/ - modules
 
-## Instructions for manual testing
+## 5. Instructions for manual testing
 
```

Comment:
I am unsure if you missed this section or decided not to do it, as it is empty. Consider deleting it if you don't need it.